proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,224 @@
+      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: appHMIType:UNKNOWN_1, UNKNOWN_2"`
+
+  3. Request contains a parameter (with enum item value) which is a part of the structure.
+
+      Request: `SetGlobalProperties`.
+
+      Parameter `menuIcon="Image"`, sub-parameter: `imageType`, `type="ImageType"`
+
+      ```
+      menuIcon = {
+        imageType = "UNKNOWN"
+      }
+      ```
+
+      SDL Core removes the `menuIcon` parameter from the request since the parameter `imageType` is mandatory for the `ImageType` structure.
+      However, request will be processed successfully since the `menuIcon` parameter is optional for `SetGlobalProperties` request and the following response will be provided to mobile application:
+
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: menuIcon.imageType:UNKNOWN"`
+
+  4. Request contains a parameter which is an array of elements and each element has a parameter which value is an enum item.
+
+      Request: `Show`.
+
+      Parameter: `softButtons`, `type="SoftButton"`, sub-parameter: `type`, `type="SoftButtonType"`.
+
+      There are 2 soft buttons provided in the request:
+
+      ```
+      softButtons = [
+        { softButtonID = 1, type = "UNKNOWN" },
+        { softButtonID = 2, type = "IMAGE" }
+      ]
+      ```
+
+      SDL Core removes the whole element with the button `1` from the request since the `type` parameter is mandatory for `SoftButton` structure:
+
+      ```
+      softButtons = [
+        { softButtonID = 2, type = "IMAGE" }
+      ]
+      ```
+
+      The request is successfully processed with the remaining button `2` and the following response is  provided to mobile application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: softButtons[1].type:UNKNOWN"`
+
+### Mobile libraries changes (JavaSuite and iOS)
+
+  This solution affects the mobile application in 2 ways:
+
+  1. The way the mobile application registers itself with SDL Core.
+
+  2. The way the mobile application sends requests to SDL Core.
+
+#### Changes related to mobile application connecting to SDL Core
+  To filter out unsupported items we need to know the exact protocol version, not only the major part of it but the minor part as well since certain parameters were added in between the major change. For instance, `AppHMIType` items `PROJECTION` and `REMOTE_CONTROL` were rolled in as early as 4.5. There is a possible case when we know the major is 4 but the information about the minor part of it is unknown before `RegisterAppInterface` request is sent and responded. However, even upon receipt of the response, the minor version of the protocol may still be not confirmed since both `syncMsgVersion` and `sdlVersion` are not mandatory parameters and thus can be absent in the response
```

Comment:
```suggestion
  To filter out unsupported items we need to know the exact protocol version, not only the major part of it but the minor part as well since certain parameters were added in between the major change. For instance, `AppHMIType` items `PROJECTION` and `REMOTE_CONTROL` were rolled in as early as 4.5. There is a possible case when we know the major version is 4 but the information about the minor part of it is unknown before `RegisterAppInterface` request is sent and responded. However, even upon receipt of the response, the minor version of the protocol may still be not confirmed since both `syncMsgVersion` and `sdlVersion` are not mandatory parameters and thus can be absent in the response.
```